src/dhcp/clientv4.rs

```diff
@@ -357,6 +353,7 @@ impl Client {
         net_trace!("DHCP reset");
         self.state = ClientState::Discovering;
         self.next_egress = now;
+        self.lease_expiration = None;
```

Comment:
I believe that we actually shouldn't lose track of the lease expiration date through a reset (at least, there should be a way to return to discovering without losing the expiration date of our lease) because we are allowed to continue using the IP even if the ethernet link connection is lost.